parsers/saarland.py

```diff
@@ -0,0 +1,199 @@
+
+
+def build_hours(opening_hours):
+    if not opening_hours:
+        return []
+
+    start = datetime.datetime.strptime(
+        opening_hours['start'], UTC_DATE_STRING).time()
+    end = datetime.datetime.strptime(
+        opening_hours['end'], UTC_DATE_STRING).time()
+    return [
+        'verfügbar von %s bis %s Uhr' %
+        (datetime.time.strftime(start, "%H:%M"),
+            datetime.time.strftime(end, "%H:%M"))
+    ]
+
+
+def build_location(description):
+    return ['Ort: %s' % description] if description else []
+
+
+def parse_url(url, today=False):
+    global base_data
+
+    canteen = LazyBuilder()
+    with urlopen(url) as response:
+        data = json.loads(response.read().decode())
+
+    for day in data['days']:
+        date = day.get('date')
+
+        if today and (datetime.datetime.now().date() !=
+                      datetime.datetime.strptime(date, UTC_DATE_STRING).date()):
+            continue
+
+        for counter in day['counters']:
+            counter_name = counter['displayName']
+            counter_description = counter['description']
+            counter_hours = counter.get('openingHours')
+            counter_color = counter.get('color')
+
+            for meal in counter['meals']:
+                if 'knownMealId' in meal:
+                    print('knownMealId: %s' % meal['knownMealId'])
+
+                meal_name = meal['name']
+                if 'category' in meal:
+                    meal_name = '%s: %s' % (meal['category'], meal_name)
+
+                meal_notes = (
+                    build_location(counter_description) +
+                    build_hours(counter_hours) +
+                    build_notes(meal['notices'], meal['components']))
+
+                meal_prices = {}
+                if 'prices' in meal:
+                    prices = meal['prices']
+                    for role in prices:
+                        if role in base_data['roles']:
+                            meal_prices[base_data['roles'][role]] = prices[role]
+                        else:
+                            print('Ignoring price for unknown role %s.' % role)
+
+                if 'pricingNotice' in meal:
+                    meal_notes.append(meal['pricingNotice'])
+
+                canteen.addMeal(date, counter_name,
+                                meal_name, meal_notes, meal_prices)
+
+    return canteen.toXMLFeed()
+
+
+with urlopen(URL_BASE + URL_BASE_DATA) as response:
+    data = json.loads(response.read().decode())
+base_data = {}
+base_data['locations'] = []
+base_data['roles'] = {}
+
+base_data['notices'] = data['notices']
+for loc in data['locations']:
+    base_data['locations'].append(loc)
+for role in data['priceTiers']:
+    if role not in ROLES:
+        print('New unknown price tier: %s (displayName: %s)' %
+              (role, data['priceTiers'][role]))
+        print('All prices for this price tier will be ignored.')
+        print('Please consider updating the parser!')
```

Comment:
I don't think printing these notes (also above) really helps as probably no one notices these anyway. You could throw an exception, which will get noticed, as openmensa.org sends emails when parsers fail, or simply ignore unknown roles.

Reply:
Okay, I've changed it to throw an error instead.